### PR TITLE
feat: support multiple audiences

### PR DESCRIPTION
**Pull request by peterphanouvong**
# Explain your changes
- allow passing in audiences as string of audiences separated by whitespace
- set the url params in a way to allow for more than 1 audience

_Suppose there is a related issue with enough detail for a reviewer to understand your changes fully. In that case, you can omit an explanation and instead include either “Fixes #XX” or “Updates #XX” where “XX” is the issue number._

# Checklist

- [x] I have read the [“Pull requests” section](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md#pull-requests) in the [contributing guidelines](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I agree to the terms within the [code of conduct](https://github.com/kinde-oss/.github/blob/main/.github/CODE_OF_CONDUCT.md).

🛟 _If you need help, consider asking for advice over in the [Kinde community](https://thekindecommunity.slack.com)._

### CODE REVIEW

File: lib/sdk/oauth2-flows/AuthCodeAbstract.ts [hunk 1]
```diff
@@ -275,6 +271,16 @@ export abstract class AuthCodeAbstract {
     for (const key in searchParamsObject)
       searchParams.append(key, searchParamsObject[key]);
 
+    if (this.config.audience) {
+      this.config.audience
+      .trim()
+      .split(/\s+/)
+      .forEach((aud) => {
+        urlSearchParams.append('audience', aud);
```
Comment (DaveOrDead) [inline lib/sdk/oauth2-flows/AuthCodeAbstract.ts:279]: @peterphanouvong I think you want `searchParams`

File: lib/sdk/oauth2-flows/ClientCredentials.ts [hunk 2]
```diff
@@ -115,7 +115,12 @@ export class ClientCredentials {
     });
 
     if (this.config.audience) {
-      searchParams.append('audience', this.config.audience);
+      this.config.audience
+      .trim()
+      .split(/\s+/)
+      .forEach((aud) => {
+        urlSearchParams.append('audience', aud);
```
Comment (DaveOrDead) [inline lib/sdk/oauth2-flows/ClientCredentials.ts:122]: @peterphanouvong I think you want `searchParams`

**Review by DaveOrDead — CHANGES_REQUESTED** [review]

**Comment by coel** [comment]
@peterphanouvong, I'd suggest instead of doing a string split convert the property `config.audience` to be an array.  It could be `string` or `string[]` to maintain backwards compatibility.

So for `OAuth2FlowOptions`, the property becomes:

```  audience?: string | string[];```

Then check later if it is an array or not.  Could be a condition or do something to make an array if not like:

```      const audienceArray = Array.isArray(this.config.audience) ? this.config.audience : [this.config.audience];```

**Comment by DaveOrDead** [comment]
@coel I guess a lot of the time the audience is being passed in via a `.env` file which would mean declaring

```
AUDIENCE=['a','b']
```
and then having to do some nastiness to convert that environment variable string into an array?

**Comment by coel** [comment]
@DaveOrDead , I think if it is nasty we shouldn't force it onto users in the base SDK.  It seems like whitespace is valid in an audience, though we don't seem to handle it.  We might need to patch that up.

If we have helpers to deal with .env I think it should happen in the layer .env is introduced.

**Comment by DaveOrDead** [comment]
@coel I'm ok with that @peterphanouvong are you ok to update?

**Review by coel — APPROVED** [review]

**Review by DaveOrDead — APPROVED** [review]

**Pull request merged**